onnxruntime/core/providers/webgpu/program_manager.h

```diff
@@ -30,7 +30,7 @@ class ProgramArtifact {
   const std::vector<int> shape_uniform_ranks;
 
   ProgramArtifact(ProgramArtifact&&) = default;
-  ProgramArtifact& operator=(ProgramArtifact&&) = default;
+  ProgramArtifact& operator=(ProgramArtifact&&) = delete;  // can't change const members.
```

Comment:
Would it be safer if we disable both of them?

Reply:
program artifact is used as a member of `ProgramManager` class:
```
std::unordered_map<std::string, ProgramArtifact> programs_;
```
and move constructor is needed